cosmwasm/its.ts

```diff
@@ -67,19 +67,88 @@ export async function checkSingleTokenRegistration(
     return registered;
 }
 
-// TODO tkulik: This command will be used to get the supply of the token on the chain.
-export async function getTokenSupply(tokenAddress: string, rpc: string): Promise<string> {
-    const provider = getDefaultProvider(rpc);
-    const token = new Contract(tokenAddress, IInterchainToken.abi, provider);
-    const supply = await token.totalSupply();
-    return supply.toString();
+export async function modifyTokenSupply(
+    client: ClientManager,
+    config: ConfigManager,
+    interchainTokenServiceAddress: string,
+    tokenId: string,
+    tokenAddress: string,
+    chain: string,
+    dryRun: boolean,
+) {
+    const tokenRegistered = await checkSingleTokenRegistration(config, client, interchainTokenServiceAddress, tokenId, chain);
+    if (!tokenRegistered) {
+        printInfo(`Token ${tokenId} on ${chain} is not registered`);
+        return;
+    }
+
+    const { origin_chain } = await client.queryContractSmart(interchainTokenServiceAddress, {
+        token_config: { token_id: formatTokenId(tokenId) },
+    });
+
+    if (origin_chain === config.getChainConfig(chain).axelarId) {
+        printInfo(`Token ${tokenId} origin chain is ${chain}, it should be set to untracked.`);
+        return;
+    }
+
+    const { supply, isTokenSupplyTracked } = await getTokenInstanceInfo(tokenAddress, config.getChainConfig(chain).rpc);
```

Comment:
This assumes that the chain is an EVM chain. This method should probably check that someone doesn't call it for an unsupported chain type. To start with, it should support an EVM chain, but in the future we can add support for Sui/Solana etc. to easily manage token supply